### PR TITLE
Azure Logs: Expand the integration docs adding a section for Event Hub partitions

### PR DESCRIPTION
**Pull request by zmoog**
## Proposed commit message

Extend the Azure Logs integration docs adding a section dedicated to setting up event hub partitions.

The existing docs didn't cover event hub partitions, leaving the responsibility of reading the Microsoft documentation. This PR gives guidance on the event hub partitions, linking to Microsoft docs for advanced use cases.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] ~~I have verified that all data streams collect metrics or logs.~~
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] ~~I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).~~

## Related issues

- https://github.com/elastic/integrations/issues/10557

### CODE REVIEW

File: packages/azure/_dev/build/docs/README.md [hunk 1]
```diff
@@ -129,6 +129,81 @@ If you are familiar with Kafka, here's a conceptual mapping between the two:
...
+
+└ Event Hub ─ ─ ─ ─ ─ ─ ─ ┘    └ Agent ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘
+```
+
+
+##### Two or more Agents
+
+With more than one Agent, setting the number of partitions is crucial. The agents share the existing partitions to scale out performance and improve availability.
+
+The number of partitions must at least match the number of agents.
```
Comment (efd6) [inline packages/azure/_dev/build/docs/README.md:172]:
```suggestion
The number of partitions must be at least the number of agents.
```
"at least match" becomes non-ordinal, so it's unclear.

File: packages/azure/_dev/build/docs/README.md [hunk 2]
```diff
@@ -129,6 +129,81 @@ If you are familiar with Kafka, here's a conceptual mapping between the two:
 | Consumer Group | Consumer Group    |
 | Offset         | Offset            |
 
+
+#### How many partitions?
+
+The number of partitions is essential to balance Event Hub cost and performance.
```
Comment (muthu-mps) [inline packages/azure/_dev/build/docs/README.md:135]: As we represent the `Event Hub` in different cases in the multiple places. Some places we do have the `Event hub`. To make this consistent, I would recommend to keep lower case in the descriptions and the titles as `Event Hub`.

File: packages/azure/_dev/build/docs/README.md [hunk 3]
```diff
@@ -129,6 +129,81 @@ If you are familiar with Kafka, here's a conceptual mapping between the two:
...
+                                  └─────────────────┘
+│                         │    │                         │
+
+└ Event Hub ─ ─ ─ ─ ─ ─ ─ ┘    └ Agent ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘
+```
+
+
+##### Recommendations
+
+Create an event hub with at least two partitions. Two partitions allow low-volume deployment to support high availability with two agents. Please consider creating four partitions or more to handle medium-volume deployments with availability.
```
Comment (alaudazzi) [inline packages/azure/_dev/build/docs/README.md:201]:
```suggestion
Create an event hub with at least two partitions. Two partitions allow low-volume deployment to support high availability with two agents. Consider creating four partitions or more to handle medium-volume deployments with availability.
```

File: packages/azure/_dev/build/docs/README.md [hunk 4]
```diff
@@ -129,6 +129,81 @@ If you are familiar with Kafka, here's a conceptual mapping between the two:
...
+
+To learn more about Event Hub partitions, you can read an in-depth guide from Microsoft at https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create.
```
Comment (alaudazzi) [inline packages/azure/_dev/build/docs/README.md:203]:
```suggestion
To learn more about Event Hub partitions, read an in-depth guide from Microsoft at https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create.
```

File: packages/azure/_dev/build/docs/README.md [hunk 5]
```diff
@@ -129,6 +129,81 @@ If you are familiar with Kafka, here's a conceptual mapping between the two:
...
+
+To learn more about Event Hub partition from the performance perspective, you can read the scalability-focused document at https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-scalability#partitions.
```
Comment (alaudazzi) [inline packages/azure/_dev/build/docs/README.md:205]:
```suggestion
To learn more about Event Hub partition from the performance perspective, check the scalability-focused document at https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-scalability#partitions.
```

File: packages/azure/docs/README.md [hunk 6]
```diff
@@ -129,6 +129,81 @@ If you are familiar with Kafka, here's a conceptual mapping between the two:
...
+                                  └─────────────────┘
+│                         │    │                         │
+
+└ Event Hub ─ ─ ─ ─ ─ ─ ─ ┘    └ Agent ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘
+```
+
+
+##### Recommendations
+
+Create an event hub with at least two partitions. Two partitions allow low-volume deployment to support high availability with two agents. Please consider creating four partitions or more to handle medium-volume deployments with availability.
```
Comment (alaudazzi) [inline packages/azure/docs/README.md:201]:
```suggestion
Create an event hub with at least two partitions. Two partitions allow low-volume deployment to support high availability with two agents. Consider creating four partitions or more to handle medium-volume deployments with availability.
```

File: packages/azure/docs/README.md [hunk 7]
```diff
@@ -129,6 +129,81 @@ If you are familiar with Kafka, here's a conceptual mapping between the two:
...
+
+To learn more about Event Hub partitions, you can read an in-depth guide from Microsoft at https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create.
```
Comment (alaudazzi) [inline packages/azure/docs/README.md:203]:
```suggestion
To learn more about Event Hub partitions, read an in-depth guide from Microsoft at https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create.
```

File: packages/azure/docs/README.md [hunk 8]
```diff
@@ -129,6 +129,81 @@ If you are familiar with Kafka, here's a conceptual mapping between the two:
...
+
+To learn more about Event Hub partition from the performance perspective, you can read the scalability-focused document at https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-scalability#partitions.
```
Comment (alaudazzi) [inline packages/azure/docs/README.md:205]:
```suggestion
To learn more about Event Hub partition from the performance perspective, read the scalability-focused document at https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-scalability#partitions.
```

**Review by alaudazzi — APPROVED** [review]
Left a few minor editing suggestions, other LGTM.
More generally, you might want to check if _event hub_ is capitalized consistently.

File: packages/azure/_dev/build/docs/README.md (hunk 2 above)
Comment (zmoog) [reply packages/azure/_dev/build/docs/README.md:135]: Good point. Let me review the doc to improve consistency!

Comment (zmoog) [reply packages/azure/_dev/build/docs/README.md:135]: After reading the following docs:

- https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-about
- https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-features

@alaudazzi @muthu-mps, I'm reviewing the `_dev/docs/README.md` following these rules:

- "Event Hubs" for the service name
- "event hub" for the actual event hub

Comment (zmoog) [reply packages/azure/_dev/build/docs/README.md:135]: After reading https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/diagnostic-settings, I'm also updating the capitalization of diagnostic settings.

Comment (zmoog) [reply packages/azure/_dev/build/docs/README.md:135]: I'm not 100% sure about how to capitalize the names in the diagrams. @alaudazzi, let me know what you think!

File: packages/azure/_dev/build/docs/README.md [hunk 9]
```diff
@@ -129,7 +129,82 @@ If you are familiar with Kafka, here's a conceptual mapping between the two:
...
+
+└ Event Hub ─ ─ ─ ─ ─ ─ ─ ┘    └ Agent ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘
+```
+
+
+##### Two or more Agents
+
+With more than one Agent, setting the number of partitions is crucial. The agents share the existing partitions to scale out performance and improve availability.
+
+The number of partitions must be at least the number of agents.
```
Comment (tetianakravchenko) [inline packages/azure/_dev/build/docs/README.md:172]: looking to the diagram below - is it possible that first agent when starting create workers of all partitions, so when second agent starts - there are no "free" partitions? 
correctly I understand that it is not possible that workers of different agents are picking the same partition?

Comment (zmoog) [reply packages/azure/_dev/build/docs/README.md:172]: The agents use blobs in a storage account to share two kinds of information: partition ownership [^1] and checkpoint information.

For example, when the input starts:

- it leases the blob for the partition 
- reads the checkpoint info
- and sets the ownership. 

If there's another input from another agent using the same agent policy, this input knows that another input claimed that partition and moves to the next partition.

The processor v2 (thanks to the modern event hub SDK) offers a configurable [load balancing strategy](https://github.com/zmoog/beats/blob/85e265cfe99425cbf3adf307a73f5a079b348d80/x-pack/filebeat/input/azureeventhub/v2_input.go#L282-L302) for partition claiming.

From [sdk/messaging/azeventhubs/processor.go](https://github.com/Azure/azure-sdk-for-go/blob/246ab6cc71f454bbc4277fafdd31e9b6e763e320/sdk/messaging/azeventhubs/processor.go#L24-L35);

> ProcessorStrategyBalanced will attempt to claim a single partition at a time, until each active owner has an equal share of partitions.


[^1]: The implementation details differ between processors v1 and v2. Processor v1 stores the ownership information in a JSON document, while processor v2 stores it as blob metadata.

Comment (tetianakravchenko) [reply packages/azure/_dev/build/docs/README.md:172]: Thanks for the detailed explanation!

**Review by muthu-mps — APPROVED** [review]
LGTM!

**Pull request merged**